libraries/Microsoft.Bot.Builder/BotAdapter.cs

```diff
@@ -224,6 +225,22 @@ protected async Task RunPipelineAsync(ITurnContext turnContext, BotCallbackHandl
             // Call any registered Middleware Components looking for ReceiveActivityAsync()
             if (turnContext.Activity != null)
             {
+                if (turnContext.Activity.Locale != null)
+                {
+                    try
+                    {
+                        Thread.CurrentThread.CurrentCulture = new CultureInfo(turnContext.Activity.Locale);
+                        (turnContext as TurnContext).Locale = turnContext.Activity.Locale;
+                    }
+#pragma warning disable CA1031 // Do not catch general exception types
+                    catch
+                    {
+                        // if turnContext.Activity.Locale is illegal, then TurnContext.Locale will set to en-US as default. 
+                        (turnContext as TurnContext).Locale = "en-US";
+                    }
```

Comment:
Do not hardcode en-US, default to the servers or machine CurrentCulture instead

Reply:
change to the Thread.CurrentThread.CurrentCulture.Name